holoviews/core/spaces.py

```diff
@@ -788,6 +804,105 @@ def dynamic_redim(obj):
+            kdims = self.kdims+dimensions
+
+            # Define callback
+            def collation_cb(*args, **kwargs):
+                return self[args[:self.ndims]][args[self.ndims:]]
+            callback = Callable(collation_cb, inputs=[self])
+
+            return self.clone(shared_data=False, callback=callback,
+                              kdims=kdims, streams=streams)
+        elif isinstance(self.last, (Layout, NdLayout, GridSpace)):
+            # Expand Layout/NdLayout
+            from ..util import Dynamic
+            new_item = self.last.clone(shared_data=False)
+
+            # Get stream mapping from callback
+            remapped_streams = []
+            streams = self.callback.stream_mapping
+            for i, (k, v) in enumerate(self.last.data.items()):
+                vstreams = streams.get(i, [])
+                if not vstreams:
+                    if isinstance(self.last, Layout):
+                        for l in range(len(k)):
+                            path = '.'.join(k[:l])
+                            if path in streams:
+                                vstreams = streams[path]
+                                break
+                    else:
+                        vstreams = streams.get(k, [])
+                if any(s in remapped_streams for s in vstreams):
+                    raise ValueError(
+                        "The stream_mapping supplied on the Callable "
+                        "is ambiguous please supply more specific Layout "
+                        "path specs.")
+                remapped_streams += vstreams
+
+                # Define collation callback
+                def collation_cb(*args, **kwargs):
+                    return self[args][kwargs['collation_key']]
+                callback = Callable(partial(collation_cb, collation_key=k),
+                                    inputs=[self])
+                vdmap = self.clone(callback=callback, shared_data=False,
+                                   streams=vstreams)
+
+                # Remap source of streams
+                for stream in vstreams:
+                    if stream.source is self:
+                        stream.source = vdmap
+                new_item[k] = vdmap
+
+            unmapped_streams = [repr(stream) for stream in self.streams
+                                if (stream.source is self) and
+                                (stream not in remapped_streams)
+                                and stream.linked]
+            if unmapped_streams:
+                raise ValueError(
+                   'The following streams are set to be automatically '
+                   'linked to a plot, but no stream_mapping specifying '
+                   'which item in the (Nd)Layout to link it to was found:\n%s'
+                    % ', '.join(unmapped_streams)
+                )
+            return new_item
+        else:
+            self.warning('DynamicMap does not need to be collated.')
+            return dmap
```

Comment:
Why warn? 

It isn't inefficient to call collate pointlessly in such a case so just return the DynamicMap as it is already ok.